### ISSUE TITLE
Lost atom error - on trained data

### ISSUE THREAD

**Issue by vsumaria** [MDIL-SNU/SIMPLE-NN_v2#77]
From my previous issue, where I was trying to train a HDNNP with 5 elements. I was able to generate a preliminary rough NNP, but am facing a small issue when validating the potential (both for the training set itself as well as the validation set). 

Few of the structures, that are in the training set give me a "Lost Atom" error when I try to validate them with lammps.  Here's the parallel script I use with ase and joblib to quickly do "single-point" calculations with the DFT reference data stores in train.traj. 

```
from ase.io import *
from ase.calculators.lammpslib import LAMMPSlib
import matplotlib.pyplot as plt
import numpy as np
from joblib import Parallel, delayed

cmds = ["pair_style nn",
        "pair_coeff * * potential_saved_bestmodel Al H Pt C O"]
lammps = LAMMPSlib(lmpcmds=cmds, atom_types= {'Al':1, 'H':2, 'Pt':3 ,'C':4 ,'O':5}, keep_alive='False', log_file='test.log')

traj = read('train.traj',':')
collect=[atoms for i,atoms in enumerate(traj) if i%5==0]
edft = [atoms.get_potential_energy()/len(atoms) for atoms in collect]
np.savetxt('edft.txt',edft)

def energy(i,atoms):
    ed = atoms.get_potential_energy()/len(atoms)
    atoms.calc = lammps
    en = atoms.get_potential_energy()/len(atoms)
    return en

enn = Parallel(n_jobs=24, backend='multiprocessing')(delayed(energy)(i,atoms) for i,atoms in enumerate(collect))
np.savetxt('enn.txt',enn)
```

Initially, my script was getting struck, which I later realized was due to the lost atoms error from lammps which I was able to correct, but then I see that the error on the structures that show this lost atoms error is extremely high of the order of 1 eV/atom.

Can you help me understand why a structure in the training set would give Lost atoms error when validating ?

**Comment by vsumaria** [comment]
Ok, seems like an ase LAMMPSlib issue and not LAMMPS. 

This works fine: 

```
from ase.io import *
import re, os
from lammps import PyLammps
from ase.calculators.lammpslib import LAMMPSlib
import matplotlib.pyplot as plt
import numpy as np
from joblib import Parallel, delayed
from pymatgen.io.ase import AseAtomsAdaptor
from pymatgen.io.lammps.data import LammpsData

cmds = ["pair_style nn",
        "pair_coeff * * potential_saved_bestmodel Al Pt H O C"]
lammps = LAMMPSlib(lmpcmds=cmds, atom_types= {'Al':1, 'H':2, 'Pt':3 ,'C':4 ,'O':5}, keep_alive='False', log_file='test.log')

traj = read('train.traj',':')
collect=[atoms for i,atoms in enumerate(traj)]
edft = [atoms.get_potential_energy()/len(atoms) for atoms in collect]
np.savetxt('edft.txt',edft)

def energy(i,atoms):

    n = len(atoms)
    ase_adap = AseAtomsAdaptor()
    atoms_ = ase_adap.get_structure(atoms)
    ld = LammpsData.from_structure(atoms_,atom_style='atomic')
    ld.write_file('struc/struc'+str(i)+'.data')

    L = PyLammps()
    L.command("units  metal")
    L.command("dimension  3")
    L.command("atom_modify map yes")
    L.command("box tilt large")
    L.command("boundary  p p f")
    L.command("atom_style  atomic")
    L.command("read_data  struc/struc"+str(i)+".data")
    L.command("pair_style nn")
    L.command("pair_coeff * * potential_saved_bestmodel Al H Pt C O")
    L.command("log log"+str(i)+".lammps")
    L.command("thermo 1")
    L.command("thermo_style custom step pe")
    L.command("run 0")

    f = open("log"+str(i)+".lammps","r")
    Lines = f.readlines()
    patten= r'(\d+\s+\-+\d*\.?\d+)'
    e_pot = []
    for i,line in enumerate(Lines):
        s = line.strip()
        match= re.match(patten, s)
        if match != None:
            D = np.fromstring(s, sep=' ')
            e_pot.append(D[1])
    en = e_pot[0]/n

    return en

enn = Parallel(n_jobs=48, backend='multiprocessing')(delayed(energy)(i,atoms) for i,atoms in enumerate(collect))
np.savetxt('enn.txt',enn)

print(edft)
print(enn)

```
